app/utils/wellknown.js

```diff
@@ -0,0 +1,110 @@
+// thanks to npm well-known-wallets-hns 
+import https from "https";
+import hdns from "hdns";
+
+const init = hsd => hdns.setServers(['103.196.38.38', '103.196.38.39', '103.196.38.40'] );
+
+class TLSAError extends Error {
+    constructor(message = 'Invalid TLSA', code = 'EINSECURE') {
+        super(message);
+        this.code = code;
+    }
+}
+
+const verifyTLSA = async (socket, host) => {
+    const cert = socket.getPeerCertificate(false);
+    const tlsa = await hdns.resolveTLSA(host, 'tcp', 443);
+
+    try {
+        const valid = hdns.verifyTLSA(tlsa[0], cert.raw);
+
+        if (!valid) {
+            throw new TLSAError();
+        }
+    } catch (e) {
+        throw new TLSAError();
+    }
+};
+
+const getAddress = (host, { dane = false, ca = !dane, token = 'HNS' } = {}) =>
+    new Promise(async (resolve, reject) => {
+        const options = {
+            rejectUnauthorized: ca,
+        };
+
+        const req = https.get(`https://${host}/c${token}`, options, res => {
```

Comment:
Missing path?

Reply:
Yes. Make a bug just make a commit and change the wellknown address validate path to daneOrCa.

Reply:
@Falci Sorry for my wrong operation. I clicked the resolve conversation. Seems the comment was removed here. I mean I can make up to feature on the setting page.  
1. Config the strategy to validate well-known address
2. config the hdns server used to, default use hdns.io.

Reply:
How does hdns.io work? They aren't signing records so there's no way to verify that the TLSA is correct.

Reply:
@rithvikvibhu hdns.io  is a handshake name resolver that can get tlsa record of one name through the ns server you set on to the handshake blockchain.

Reply:
True, but to be able to trust the TLSA record is correct, the resolver should sign it (DNSSEC). Even if the nameserver is picked up from the blockchain, the address can be at any depth (abc.def.efg.handshakename) so it will need recursive resolving.
That's why it's not possible to get HTTP/DANE with https://hdns.io currently, because the TLSA record cannot be trusted.

Reply:
@rithvikvibhu the resolver doesnt sign TLSA, the zone owner would and they just serve the RRSIG from their nameserver. It's the resolvers job to verify all the signatures. As far as I can tell, hdns' recursive resolver does verify dnssec (`ad` bit is set):

```
--> dig @103.196.38.38 _443._tcp.proofofconcept tlsa +dnssec
...
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1
...
;; ANSWER SECTION:
_443._tcp.proofofconcept. 3600	IN	TLSA	3 1 1 22E3C95A736E370FA38E7D94239F49C7A9EF961AF94E05AE8CC74FC3 CA2BA5CE
_443._tcp.proofofconcept. 86400	IN	RRSIG	TLSA 8 3 3600 20220225162619 20210224162619 58608 proofofconcept. gCIZIIm6KkUDDee4kqmuEjZ9hhTUPgoORUKwgTrzlIVV0p+4B83duHWV S/amZfpNwM0vWJ25If9UstG/q6QI1PrbJuSoonKl9eXaAihbZ1csfV4Q 7hC2JyrlarAsv7VjvnQwc31DYxywwOCfLuV5aeo1CFLrtsOGgnG6eJXp MXb79351FqBS2xVUvNyZfMLFcVq5xgAenH3JP5KwuOsBgNw+1mtLBg8N DQVQaIlPz/9/l9wrDqhuzhciOhFkHHZcq/cD/W1xsEgTJow3WdaIbAs1 FTy7CwowPeiYLD9RCnTYMv66cdCEvqKUgIHa2ib9k9d0D8zmoQNhIxHJ H9fSDQ==
```

However I do not think their DoH service does dnssec, I'm still trying to figure that out.

Reply:
Ahh, my bad, got confused there. You're right @pinheadmz and @v1xingyue.

Just tried it and looks like it only works for handshake domains:

Has RRSIG:
```sh
dig @1.1.1.1 _443._tcp.torproject.org TLSA +dnssec
```

Doesn't have RRSIG:
```sh
dig @103.196.38.38 _443._tcp.torproject.org TLSA +dnssec
```

Reply:
What?! that is fucking bonkers